src/shogun/multiclass/BruteKNNSolver.cpp

```diff
@@ -6,21 +6,28 @@
 #include <shogun/base/progress.h>
 #include <shogun/lib/Signal.h>
 #include <shogun/multiclass/BruteKNNSolver.h>
+#include <shogun/multiclass/KNNSolver.h>
+#include <shogun/machine/DistanceMachine.h>
+
+#include <iostream>
+#include <algorithm>
 
 using namespace shogun;
 
-BruteKNNSolver::BruteKNNSolver(const int32_t k, const float64_t q, const int32_t num_classes, const int32_t min_label, const SGVector<int32_t> train_labels, const SGMatrix<index_t> NN):
+BruteKNNSolver::BruteKNNSolver(const int32_t k, const float64_t q, const int32_t num_classes, const int32_t min_label, const SGVector<int32_t> train_labels):
 KNNSolver(k, q, num_classes, min_label, train_labels)
 {
+	std::cout<<"entered BruteKNNSolver::constructor\n";
```

Comment:
this is a classic candidate for a SG_DEBUG....although one wouldnt see it in but maybe another helper method